src/components/App/App.tsx

```diff
@@ -7,7 +7,28 @@ import { PokemonList } from '../PokemonList/PokemonList';
 import { getData } from '../../actions';
 import { Wrapper, Header, Title, Logo, LoadingWrapper } from './App.s';
 
-class AppPure extends React.Component <any> {
+type StateProps = {
+  state: {
+    isLoading: boolean,
+    isError: boolean,
+    data: object[],
+    selectedPokemon: {
```

Comment:
ten typ Pokemona powtarza się w wielu miejsach, dlatego ten i podobne typy powinienes wynieść do osobnego modułu, żebys mógł go użyć wielu miejscach
w /src dodaj katalog /model, który bedzie przechowywał modele, pierwszy z nich to oczywiscie Pokemon, tworzysz po prostu plik Pokemon.ts w którym eksportujesz 
`epxort type Pokemon = Readonly<{ //tutaj jeogpola }> `
- Readonly dlatego, że nie chcemy żeby jakiś model został nadpisany, ponieważ możemy to zrobic tylko przez reduxa, a nie w reactcie, wiec dzięki temu TS bedzie nas pilnował